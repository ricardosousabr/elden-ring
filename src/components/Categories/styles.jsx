import styled from 'styled-components'

export const BoxButton = styled.div`
  height: 600px;
  width: 750px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
`
