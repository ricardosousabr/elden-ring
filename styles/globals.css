* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

body {
  max-width: 100vw;
  background: url(/img/img-elden-ring-background.jpg) no-repeat;
  background-size: 1920px 970px;
}