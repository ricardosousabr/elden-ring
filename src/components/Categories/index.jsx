import Link from 'next/link'
import Button from '../Button'
import { BoxButton } from './styles'
import axios from 'axios'
import { useState } from 'react'

export default function Categories() {
  const [responseApi, setResponseApi] = useState(null)
  const categoriesName = [
    'Ammos',
    'Armors',
    'Ashes of War',
    'Bosses',
    'Classes',
    'Creatures',
    'Incantations',
    'Items',
    'Locatiosn',
    'NPCs',
    'Shields',
    'Sorceries',
    'Spirits',
    'Talismans',
    'Weapons',
  ]

  function getData() {
    axios
      .get('https://eldenring.fanapis.com/api/armors?name=All-knowing%20Armor')
      .then((response) => {
        console.log(response.data)
        setResponseApi(response.data)
      })
  }

  return (
    <BoxButton>
      {categoriesName.map((categorie, index) => {
        return (
          <Link href="/subcategory" key="link">
            <Button key={index}>{categorie}</Button>
          </Link>
        )
      })}
    </BoxButton>
  )
}
