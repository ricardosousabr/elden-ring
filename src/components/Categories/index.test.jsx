/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import Categories from './index'

import { jest } from '@jest/globals'

jest.mock('next/link', () => {
  return ({ children }) => {
    return children
  }
})

it('renders homepage unchanged', () => {
  const { container } = render(<Categories />)
  expect(container).toMatchSnapshot()
})
