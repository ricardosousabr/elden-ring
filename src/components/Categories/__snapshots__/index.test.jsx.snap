// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="styles__BoxButton-sc-c4zca5-0 iphxnP"
  >
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Ammos
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Armors
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Ashes of War
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Bosses
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Classes
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Creatures
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Incantations
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Items
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Locatiosn
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        NPCs
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Shields
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Sorceries
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Spirits
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Talismans
      </button>
    </a>
    <a
      href="/subcategory"
    >
      <button
        class="styles__ButtonCategorie-sc-ufcdqw-0 cbhaIi"
      >
        Weapons
      </button>
    </a>
  </div>
</div>
`;
